docs/operate/continuous-integration-continuous-deployment/circleci/index-circleci.mdx

```diff
@@ -0,0 +1,236 @@
+
+
+Choose your preferred login method. To make it easier, let us choose “GitHub” for this demonstration.
+
+
+## Step 2. Verify your permission on GitHub
+
+
+
+
+![alt_text](images/circleci-auth.png "image_tooltip")
+
+
+
+## Step 3. Select your project repository and click “Setup Project”
+
+We will be using the Redis rate-limiting project for this demonstration.
+
+
+
+
+![alt_text](images/circleci-proj.png "image_tooltip")
+
+
+
+## Step 4. Create a new CircleCI configuration file
+
+CircleCI believes in _configuration as code_. As a result, the entire delivery process from build to deploy is orchestrated through a single file called config.yml. The config.yml file is located in a folder called .circleci at the top of your project. CircleCI uses the YAML syntax for config.
+
+
+
+
+![alt_text](images/circleci-config.png "image_tooltip")
+
+
+As we haven’t yet created a config.yml file, let’s choose the “Fast” option to create a new config.yml based on the available template that is editable.
+
+
+
+
+![alt_text](images/circleci-config2.png "image_tooltip")
+
+
+Once you click “Set Up project,” it will ask you to select sample configs as shown in the following screenshot:
+
+
+
+
+![alt_text](images/circleci-demoapp.png "image_tooltip")
+
+
+Add the following content under .circleci/config.yml and save the file:
+
+
+```
+version: 2.1
+orbs:
+  heroku: circleci/heroku@1.2.6
+workflows:
+  heroku_deploy:
+    jobs:
+      - heroku/deploy-via-git
+      
+     
+
+
+```
+
+
+In the configuration above, we pull in the Heroku orb circleci/heroku@1.2.6, which automatically gives us access to a powerful set of Heroku jobs and commands.
+
+One of those jobs is the heroku/deploy-via-git, which deploys your application straight from your GitHub repo to your Heroku account.
+
+
+## Step 5. Merge the pull request
+
+Once you make the new change, it will ask you to raise a new PR. Go ahead and merge the changes as of now.
+
+
+
+
+![alt_text](images/circleci-merge.png "image_tooltip")
+
+
+
+## Step 6. Set up Heroku account
+
+
+## [Follow these steps ](https://developer.redis.com/create/heroku/portal)to set up a Heroku account and create a new app `rate-limit-python`. You will need the Heroku API key for this demo.
+
+
+## 
+
+
+![alt_text](images/heroku-setup.png "image_tooltip")
+
+
+
+## Configuring Heroku access on CircleCI
+
+
+## Before you push your project to Heroku from CircleCI, you will need to configure an authenticated handshake between CircleCI and Heroku. Configure the handshake by creating two [environment variables](https://circleci.com/docs/2.0/env-vars/) in the settings for your CircleCI project:
+
+
+
+* HEROKU_APP_NAME is the name of your Heroku application (in this case, simple-node-api-circleci)
+* HEROKU_API_KEY is your Heroku account API key. This can be found under the Account tab of your Heroku account under Account Settings. Scroll to the API Key section and click **Reveal** to copy your API key.
```

Comment:
```suggestion
* `HEROKU_API_KEY` is your Heroku account API key. This can be found under the Account tab of your Heroku account under Account Settings. Scroll to the API Key section and click **Reveal** to copy your API key.
```